File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.java
import java.util.*;
class Solution {
    public int solution(int[][] lines) {
        int answer = 0;
        int arr[] = new int[201];
        for(int i=0; i<lines.length; i++){
            for(int j=0; j<arr.length; j++){
                if(lines[i][0] +100 < j && lines[i][1] +100 >= j){
                    arr[j]++;
                }
            }
        }
        for(int i : arr){
            if(i>=2){
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.js
function solution(lines) {
    var answer = 0;
    let arr = new Array(201).fill(0);
    for(let i=0; i<lines.length; i++){
        for(let j = lines[i][0]; j<lines[i][1]; j++){
            arr[100+j]+=1;
        }
    }
    
    for(let i=0; i<arr.length; i++){
        if(arr[i] >=2){
            answer+=1
        }
    }
    return answer;
}